opensrp-reveal/src/main/java/org/smartregister/reveal/fragment/AvailableOfflineMapsFragment.java

```diff
@@ -0,0 +1,253 @@
+import org.smartregister.domain.LocationProperty;
+import org.smartregister.reveal.R;
+import org.smartregister.reveal.adapter.AvailableOfflineMapAdapter;
+import org.smartregister.reveal.contract.AvailableOfflineMapsContract;
+import org.smartregister.reveal.contract.OfflineMapDownloadCallback;
+import org.smartregister.reveal.model.OfflineMapModel;
+import org.smartregister.reveal.presenter.AvailableOfflineMapsPresenter;
+import org.smartregister.reveal.task.FileHttpServerTask;
+import org.smartregister.reveal.util.Constants;
+import org.smartregister.reveal.util.OfflineMapHelper;
+import org.smartregister.util.DateTimeTypeConverter;
+import org.smartregister.util.PropertiesConverter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AvailableOfflineMapsFragment extends BaseOfflineMapsFragment implements AvailableOfflineMapsContract.View {
+
+    private RecyclerView offlineMapRecyclerView;
+
+    private AvailableOfflineMapAdapter adapter;
+
+    private AvailableOfflineMapsPresenter presenter;
+
+    private List<OfflineMapModel> offlineMapModelList = new ArrayList<>();
+
+    private List<Location> operationalAreasToDownload = new ArrayList<>();
+
+    private OfflineMapDownloadCallback callback;
+
+    private Button btnDownloadMap;
+
+
+    public static AvailableOfflineMapsFragment newInstance(Bundle bundle) {
+
+        AvailableOfflineMapsFragment fragment = new AvailableOfflineMapsFragment();
+        if (bundle != null) {
+            fragment.setArguments(bundle);
+        }
+
+        fragment.setPresenter(new AvailableOfflineMapsPresenter(fragment));
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (presenter == null) {
+            presenter = new AvailableOfflineMapsPresenter(this);
+        }
+        btnDownloadMap = null;
+
+        new FileHttpServerTask(getContext()).execute();
+
+
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_offline_map, container, false);
+        setUpViews(view);
+        initializeAdapter();
+        return view;
+    }
+
+    private void setUpViews(View view) {
+        offlineMapRecyclerView = view.findViewById(R.id.offline_map_recyclerView);
+
+        btnDownloadMap = view.findViewById(R.id.download_map);
+
+        btnDownloadMap.setOnClickListener(onClickListener);
+
+    }
+
+    private void initializeAdapter() {
+        adapter = new AvailableOfflineMapAdapter(this.getContext(), onClickListener);
+        offlineMapRecyclerView.setAdapter(adapter);
+        if (offlineMapModelList != null) {
+            setOfflineMapModelList(offlineMapModelList);
+        }
+
+    }
+
+    private View.OnClickListener onClickListener = new View.OnClickListener(){
```

Comment:
implement `View.OnClickListener` on this class so that no need to a variable so that just become onClick